codecov_cli/services/staticanalysis/__init__.py

```diff
@@ -96,36 +103,42 @@ async def run_analysis_entrypoint(
         for el in response_json["filepaths"]
         if (el["state"].lower() == "created" or should_force)
     ]
+
     if files_that_need_upload:
         uploaded_files = []
-        failed_upload = []
+        failed_uploads = []
         with click.progressbar(
             length=len(files_that_need_upload),
-            label="Uploading files",
+            label=f"Upload info to storage",
         ) as bar:
-            async with httpx.AsyncClient() as client:
+            # It's better to have less files competing over CPU time when uploading
+            # Especially if we might have large files
+            limits = httpx.Limits(max_connections=20)
+            # Because there might be too many files to upload we will ignore most timeouts
+            timeout = httpx.Timeout(read=None, pool=None, connect=None, write=10.0)
```

Comment:
Not having any sort of read or connect timeout seems potentially not great but I guess in the worst case the task itself just times out.  That seems fine - we're not really blocking anything indefinitely.